src/poetry/utils/helpers.py

```diff
@@ -94,16 +94,41 @@ def download_file(
 ) -> None:
     import requests
 
+    from poetry.puzzle.provider import Indicator
+
     get = requests.get if not session else session.get
 
     response = get(url, stream=True)
     response.raise_for_status()
 
+    set_indicator = False
+    if "Content-Length" in response.headers:
+        try:
+            total_size = int(response.headers["Content-Length"])
+        except ValueError:
+            total_size = 0
+
+        fetched_size = 0
+        last_percent = 0
+
+        Indicator.set_context(f"Downloading {url}")
+        # if less than 1MB, we simply show that we're downloading but skip the updating
+        set_indicator = total_size > 1024 * 1024
+
     with open(dest, "wb") as f:
         for chunk in response.iter_content(chunk_size=chunk_size):
             if chunk:
                 f.write(chunk)
 
+                if set_indicator:
+                    fetched_size += len(chunk)
+                    percent = (fetched_size * 100) // total_size
+                    if percent > last_percent:
+                        last_percent = percent
+                        Indicator.set_context(f"Downloading {url} {percent:3}%")
+
+    Indicator.set_context(None)
```

Comment:
We should probably put this into a context manager to ensure it gets always reset. How does this behave when parallel downloads are in progress?

Reply:
I'll see if I can do a context manager in a nice pattern. 

I wasn't aware that parallel DLs could happen; that complicates things a lot. Can you give an example for when that happens and I'll investigate?

Reply:
Just an educated guess (not verified):

I don't think that parallel downloads happen during dependency resolution (afaik dependency resolution is sequential). I could imagine that parallel downloads can happen during installation (if the packages are not yet cached) because multiple packages can be installed in parallel. Even though the `download_file` function is used outside of dependency resolution, the `Indicator` seems to be only used during dependency resolution.

Reply:
Yeah; I couldn't see any paths where that would happen. I'm back at my computer now after easter so will take another check tonight if I can find time.